library/src/main/java/com/chuckerteam/chucker/internal/support/FileFactory.kt

```diff
@@ -1,8 +1,29 @@
 package com.chuckerteam.chucker.internal.support
 
 import java.io.File
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.util.concurrent.atomic.AtomicLong
 
-internal interface FileFactory {
-    fun create(): File
-    fun create(filename: String): File
+internal class FileFactory(
+    private val getFileDirectory: () -> File?
+) {
+    private val uniqueIdGenerator = AtomicLong()
+
+    fun create() = create(filename = "chucker-${uniqueIdGenerator.getAndIncrement()}")
+
+    fun create(filename: String): File? = try {
+        val directory = getFileDirectory()
+            ?: throw FileNotFoundException("Failed to create directory for temporary Chucker files")
+        File(directory, filename).apply {
+            if (exists()) {
+                delete()
+            }
+            parentFile?.mkdirs()
+            createNewFile()
```

Comment:
Shouldn't we fail fast here if this invocation returned `false`?

Reply:
I'm not sure. `createNewFile()` returns `false` only if the file already exists according to documentation. In that case, it might be more accurate to do something like this.

```kotlin
if (exists() && !delete()) {
  throw IOException("Failed to delete temporary Chucker file $file")
}
```